Workflow and makefile updates

zotoMIT opened the pull request:
Now that we have landed on a final version of workflows and makefiles for our projects, I need to do 1 more revision of the Workflows and makefile.

This sets up/refines all workflows and the makefile

It adds the prod promote workflow, which will not work until we deploy prod

This is a copy/paste from the ECR infrastructure output, this output is available to the devs in terraform cloud.

cabutlermit commented on Makefile:
```diff
@@ -1,7 +1,7 @@
 .PHONY: help install test tests update dist publish promote
```
I know that this doesn't really matter, but the last three words in the .PHONY line don't actually match up with commands in the Makefile. Do we need to worry about this?

zotoMIT commented:
OK, i removed the legacy-stage.yml workflow entirely, I dont think we're still using it for anything.  I also updated the test workflow to call the "dist-dev" command instead of the "dist" command.

hakbailey reviewed (approved):
👍
